src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -109,6 +111,13 @@ public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerial
                 {
                     return value;
                 }
+
+                // If enum contains special char, faile before pass bogus tokens into the naming policy.
+                if (s_containsSpecialChar)
```

Comment:
Is it possible for `TryParseEnumCore` to succeed when parsing the string?

Reply:
I'm not sure TBH. I would recommend adding tests that cover that particular case -- we can make a determination based on the assertions whether it is desirable behavior.

Reply:
Currentlly, the code works as below:

```fsharp
[<Flags>]
type BadEnum =
  | ``There's a comma, in my name`` = 1
  | ``There's a comma, even here`` = 2
  | ``ThisisagoodEnumValue`` = 4

// success
let enum = Enum.TryParse<BadEnum>("ThisisagoodEnumValue")

// success
let enum = JsonSerializer.Deserialize<BadEnum>(@"""ThisisagoodEnumValue""", options)

// failed
let enum = BadEnum.ThisisaoodEnumValue
let json = JsonSerializer.Serialize(enum, options)

```

Reply:
Sounds good :+1:

Reply:
Hmm, actually what about the case where we use `JsonNamingPolicy.CamelCase` and we're deserializing "thisisAgoodEnumValue"? Shouldn't we try to apply the naming policy before failing?

Reply:
Do you mean serializing?

Reply:
Do you thick it should be success when serialize Enum.ThisisAgoodEnumValue?

Reply:
See: https://github.com/dotnet/runtime/pull/76873#discussion_r997160813. I'm starting to think we should always fail if the enum contains tokens with unsupported characters.